elide-core/src/main/java/com/yahoo/elide/security/PermissionManager.java

```diff
@@ -0,0 +1,500 @@
+                || (capClassComChecks != null && capClassComChecks.length > 0)) {
+            strategy.setCommitCheck(true);
+            final List<Class<? extends Check>[]> fieldComList = new ArrayList();
+            fieldComList.add(fieldComChecks);
+            commitChecks.add(() -> {
+                fieldAwareExecute(capClassComChecks,
+                        fieldComList,
+                        capClassCheckMode,
+                        Arrays.asList(capFieldCheckMode),
+                        resource,
+                        changeSpec,
+                        false,
+                        strategy);
+                return null;
+            });
+        }
+    }
+
+    /**
+     * Execute commmit checks.
+     */
+    public void executeCommitChecks() {
+        commitChecks.forEach(Supplier::get);
+    }
+
+    private void runPermissionChecks(Class<? extends Check>[] checks,
+                                     CheckMode checkMode,
+                                     PersistentResource<?> resource,
+                                     ChangeSpec changeSpec) {
+        if (resource.getRequestScope().getSecurityMode() == SecurityMode.BYPASS_SECURITY) {
+            return;
+        }
+
+        for (Class<? extends Check> check : checks) {
+            Check handler;
+            try {
+                handler = check.newInstance();
+            } catch (InstantiationException | IllegalAccessException e) {
+                log.debug("Illegal permission check '{}' {}", check.getName(), e);
+                throw new InvalidSyntaxException("Illegal permission check '" + check.getName() + "'", e);
+            }
+
+            boolean ok = handler.ok(resource.getObject(), resource.getRequestScope(), Optional.ofNullable(changeSpec));
+
+            if (ok && checkMode == CheckMode.ANY) {
+                return;
+            }
+
+            if (!ok && checkMode == CheckMode.ALL) {
+                log.debug("ForbiddenAccess {} {}#{}", check, resource.getType(), resource.getId());
+                throw new ForbiddenAccessException();
+            }
+        }
+
+        if (checkMode == CheckMode.ANY) {
+            log.debug("ForbiddenAccess {} {}#{}", Arrays.asList(checks), resource.getType(), resource.getId());
+            throw new ForbiddenAccessException();
+        }
+    }
+
+    /**
+     * Check object and all fields for field- and class-level checks.
+     *
+     * @param classChecks Array of class/entity-level checks
+     * @param fieldChecks Array of field checks
+     * @param resource Resource
+     * @param changeSpec Change spec
+     * @param hasDeferredChecks Whether or not there are deferred checks beyond
+     * @param strategy Strategy to use upon failure/success of checks
+     * @return True if any of the field-level checks succeed, false otherwise. False if no field-level checks exist.
+     */
+    private void fieldAwareExecute(Class<? extends Check>[] classChecks,
```

Comment:
I think there should be a class that essentially represents the permission which includes the complete list of checks and the corresponding mode.  Functions should only need a single instance of this 'permission' for the class and a set of them for the set of fields that need to be checked.